### ISSUE TITLE
Feature request: support SageMaker SDK 2.0

### ISSUE THREAD

**Issue by juliensimon** [aws/aws-step-functions-data-science-sdk-python#69]
SageMaker SDK 2.0 is coming soon : https://github.com/aws/sagemaker-python-sdk/issues/1459

It would be great if the Data Science SDK could support it shortly after the release.

**Comment by cfregly** [comment]
+1.  Specifically, i'm having this issue with SageMaker SDK 2.5.5 and StepFunctions 1.1.0:

```---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-32-8cdc141c8efb> in <module>
     12     client=sfn,
     13     deploy_instance_count=deploy_instance_count,
---> 14     deploy_instance_type=deploy_instance_type,
     15 )

<ipython-input-31-298b58046d07> in __init__(self, processor, raw_input_data_s3_uri, train_split_percentage, validation_split_percentage, test_split_percentage, max_seq_length, balance_dataset, estimator, role, bucket, client, deploy_instance_count, deploy_instance_type, **kwargs)
     85             self.pipeline_name = 'training-pipeline-{date}'.format(date=self._generate_timestamp())
     86 
---> 87         self.definition = self.build_workflow_definition()
     88         self.input_template = self._extract_input_template(self.definition)
     89 

<ipython-input-31-298b58046d07> in build_workflow_definition(self)
    177             instance_type=deploy_instance_type,
    178             model=model,
--> 179             model_name=self.pipeline_name
    180         )
    181 

~/anaconda3/envs/python3/lib/python3.7/site-packages/stepfunctions/steps/sagemaker.py in __init__(self, state_id, model, model_name, instance_type, tags, **kwargs)
    203         """
    204         if isinstance(model, FrameworkModel):
--> 205             parameters = model_config(model=model, instance_type=instance_type, role=model.role, image=model.image)
    206             if model_name:
    207                 parameters['ModelName'] = model_name

AttributeError: 'TensorFlowModel' object has no attribute 'image'

```

**Comment by cfregly** [comment]
@shunjd any update on timing for SageMaker SDK 2.x support?

**Comment by brightsparc** [comment]
@cfregly I have a PR ready for merging.

**Comment by sz640** [comment]
Is this incorporated into the new release? This is also causing Pipenv not being able to lock. Specifically, when sagemaker>2.0.0 is already installed in the virtual env:

```
ERROR: Could not find a version that matches sagemaker<2.0.0,==2.15.0,>=1.71.0 (from -r /var/folders/91/xv32p1qd2g573gzb6h85ngsc0000gn/T/pipenvj8m7d8dhrequirements/pipenv-yvcfit9i-constraints.txt (line 8))
```

Pipenv won't have issues creating a new virtual env with unspecified versions of sagemaker and stepfunctions, because it'll select sagemaker==1.72.1. But this would of course break code, as mentioned in the PR above.

**Comment by wong-a** [comment]
Hey all,

We released a pre-release version (`2.0.0rc1`) of stepfunctions to PyPI that uses SageMaker 2.0 and drops Python 2 support. Since this introduces breaking changes, this is a major version bump to v2.0.

https://pypi.org/project/stepfunctions/2.0.0rc1/

We're working on a stable release after we sort out the sunsetting plan for v1 of the Step Functions Data Science SDK.

**Comment by wong-a** [comment]
v2.0.0 was released in March - which includes SageMaker SDK 2.0.

https://github.com/aws/aws-step-functions-data-science-sdk-python/releases/tag/v2.0.0
